### PR TITLE
IDEMPIERE-5643 - Error during document's workflows are not displayed on Workflow Activities form.

### PR DESCRIPTION
**Pull request by MarcosGOFavaretto**
Jira ticket: [IDEMPIERE-5643](https://idempiere.atlassian.net/browse/IDEMPIERE-5643)
# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [x] I have tested the direct scenario that my code is solving
- [x] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [x] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

# Changes
During development, I notice that the `MWFProcess.m_processMsg` field is not accessible for WWFActivity. Like showed below, there is an instance of MWFProcess, despite that, the `getProcessMsg()` does not returns the error, givin that it is another instance, where `m_processMsg` is null.

originalCode

To fix that problem and make the cited field accessible, I created a method `MWFActivity.getProcessMsg()` that returns the `MWFProcess.getProcessMsg()` method.

newMethod

Using it, the Exception can be thrown with the correct error message.

throw

# Test
Sales Order which can't be completed:
salesOrder

After user attempt to complete that document, as expected, it falls on Sales Order Workflow and waits for approve.
form

After approval, the exception is thrown at the Workflow Acitivities form.
error

[IDEMPIERE-5643]: https://idempiere.atlassian.net/browse/IDEMPIERE-5643?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Hi @MarcosGOFavaretto 

I found an error when testing this:

- as GardenUser I create a requisition for 5 PChair and ask to complete

NOTE: here I changed the MRequisition.completeIt code to fail

- as GardenAdmin I approve the requisition
- it shows the error on screen and nothing happens

- so I try then to disapprove the requisition - change the Answer to N and click OK again
- but the program doesn't take the N value - it still tries to approve it

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/wf/WWFActivity.java [hunk 1]
```diff
@@ -656,11 +657,14 @@ else if (MWFNode.ACTION_UserChoice.equals(node.getAction()))
 					m_activity.setUserChoice(AD_User_ID, value, dt, textMsg);
 					MWFProcess wfpr = new MWFProcess(m_activity.getCtx(), m_activity.getAD_WF_Process_ID(), m_activity.get_TrxName());
 					wfpr.checkCloseActivities(m_activity.get_TrxName());
+					
+					if (!Util.isEmpty(m_activity.getProcessMsg(), true))
+						throw new AdempiereException(m_activity.getProcessMsg());
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/wf/WWFActivity.java:662]: for consistency, if we add this checking here, we need to add it to the "User Action" block below too.

Comment (matheus-marcelinux) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/wf/WWFActivity.java]: @hengsin done

**Comment by matheus-marcelinux** [comment]
> Hi @MarcosGOFavaretto
> 
> I found an error when testing this:
> 
> * as GardenUser I create a requisition for 5 PChair and ask to complete
> 
> NOTE: here I changed the MRequisition.completeIt code to fail
> 
> * as GardenAdmin I approve the requisition
> * it shows the error on screen and nothing happens
> * so I try then to disapprove the requisition - change the Answer to N and click OK again
> * but the program doesn't take the N value - it still tries to approve it

Hi @CarlosRuiz-globalqss i've changed the code to not throw exception, just shows the error message in dialog, and everything worked fine

**Pull request merged**